spark/v3.3/spark/src/test/java/org/apache/iceberg/spark/source/TestDataFrameWriterV2.java

```diff
@@ -226,4 +229,100 @@ public void testWriteWithCaseSensitiveOption() throws NoSuchTableException, Pars
     fields = Spark3Util.loadIcebergTable(sparkSession, tableName).schema().asStruct().fields();
     Assert.assertEquals(4, fields.size());
   }
+
+  @Test
+  public void testMergeSchemaIgnoreDowncast() throws Exception {
+    //// test long to int ignore case
+    sql(
+        "ALTER TABLE %s SET TBLPROPERTIES ('%s'='true')",
+        tableName, TableProperties.SPARK_WRITE_ACCEPT_ANY_SCHEMA);
+
+    Dataset<Row> bigintDF =
+        jsonToDF(
+            "id bigint, data string",
+            "{ \"id\": 1, \"data\": \"a\" }",
+            "{ \"id\": 2, \"data\": \"b\" }");
+
+    bigintDF.writeTo(tableName).append();
+
+    assertEquals(
+        "Should have initial rows with long column",
+        ImmutableList.of(row(1L, "a"), row(2L, "b")),
+        sql("select * from %s order by id", tableName));
+
+    Dataset<Row> intDF =
+        jsonToDF(
+            "id int, data string",
+            "{ \"id\": 3, \"data\": \"c\" }",
+            "{ \"id\": 4, \"data\": \"d\" }");
+
+    try {
+      intDF.writeTo(tableName).option("merge-schema", "true").append();
+    } catch (IllegalArgumentException e) {
+      Assert.fail(String.format("Should not throw exception: %s", e));
+    }
+
+    assertEquals(
+        "Should include new rows with unchanged long column type",
+        ImmutableList.of(row(1L, "a"), row(2L, "b"), row(3L, "c"), row(4L, "d")),
+        sql("select * from %s order by id", tableName));
+
+    // verify the column type did not change
+    String[] tblNameSplit = tableName.split("\\.");
```

Comment:
May be easier to use the Spark Helpers here, although the validation catalog is also fine to check but I don't think it matters much in this case.

```java
    Spark3Util.loadIcebergTable(spark, tableName).schema()
```